### PR TITLE
[BUG] Book number field

### PR DESCRIPTION
**Pull request by jessiiepeng**
## Notion ticket link

[Ticket here](https://www.notion.so/uwblueprintexecs/Book-number-field-5c8bb79df1984fcba8035c33538ecae7)

## Implementation description
* changed frontend `AddStringInput` element to `AddNumberInput` element, as well as useState variables from string to number (base series number is 1) 
* change backend model property `series_order` from string to integer
* changed frontend model property `seriesOrder` from string to number
* updated sorting books by series number so it processes number instead of string 

## Steps to test
1. run migration file by following: [these instructions](https://github.com/uwblueprint/ccbc#running-migrations--seeding-database )

2. check that Book number input can only be numbers
Screen Shot 2022-05-21 at 5 11 09 PM

3. add multiple books on create review page: check that if you have multiple books, they are displayed in order of increasing book number

4. check if you can publish review of said books

## What should reviewers focus on?
* for the input fields: **(I set minimum book number to be 1, and maximum to be 1000 -> not sure what these numbers should be)**
* currently, if user doesn't enter anything, the book number (on frontend) is set to 1 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have run docker-compose up and my project compiled
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by susheelk — CHANGES_REQUESTED** [review]
Could you rebase this branch on `development`? There's an issue with editing reviews that we solved in an earlier PR that's preventing me from properly testing this. Thanks!

**Review by Saber74 — CHANGES_REQUESTED** [review]
LGTM but NaN bug upon empty book number field needs to be fixed

**Review by Saber74 — APPROVED** [review]
LGTM

- Tested locally, added multiple books, made sure NaN does not appear on empty field

File: backend/typescript/migrations/20220420005809-change_book.seriesOrder_from_string_to_int.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        type: 'INTEGER USING CAST("series_order" as INTEGER)',
+        // allowNull: true
+      }),
+    ]);
+  },
+
+  async down(queryInterface, Sequelize) {
+    return Promise.all([
+      queryInterface.changeColumn("books", "series_order", {
+        type: 'STRING USING CAST("series_order" as STRING)',
```
Comment (susheelk) [inline backend/typescript/migrations/20220420005809-change_book.seriesOrder_from_string_to_int.js:16]: Migrating down doesn't work. I don't think Postgres has a "STRING" type. Could you check what it was before (either `CHAR` or `VARCHAR`)

**Review by susheelk — CHANGES_REQUESTED** [review]

**Review by susheelk — APPROVED** [review]
LGTM! Nice work :slightly_smiling_face:

**Pull request merged**